assets/js/stripe-payment-request.js

```diff
@@ -519,13 +519,8 @@ jQuery( function( $ ) {
 				$( document.body ).trigger( 'wc_stripe_block_payment_request_button' );
 
 				// First check if product can be added to cart.
-				var addToCartButton = $( '.single_add_to_cart_button' );
-				if ( addToCartButton.is( '.disabled' ) ) {
-					if ( addToCartButton.is( '.wc-variation-is-unavailable' ) ) {
```

Comment:
I've translated this validation to the backend like the one we do when the user is trying to purchase more products than available. I think it's better to do it in the backend otherwise the user could potentially try to purchase something out of stock based solely on the information of the frontend.